x-pack/plugins/security_solution/server/lib/detection_engine/signals/search_after_bulk_create.ts

```diff
@@ -104,6 +113,79 @@ export const searchAfterAndBulkCreate = async ({
             errors: searchErrors,
           }),
         ]);
+
+        // if the result set is empty it could mean that the index
+        // has the timestamp override field mapped, but it is unused
+        // in that case we want to try to query against
+        // the @timestamp field as a backup.
+        // not only are we checking if the timestamp override field exists | not
+        // we also want to check if it is in use.
+        // we will need to figure out partial erroring later.
+        if (
+          (!isEmpty(searchErrors) || isEmpty(searchResult.hits.hits)) &&
+          ruleParams.timestampOverride != null
+        ) {
+          const {
+            searchResult: searchResultB,
+            searchDuration: searchDurationB,
+            searchErrors: searchErrorsB,
+          } = await singleSearchAfter({
```

Comment:
As a follow up PR we should look at ways to make this more efficient - the queries for primary and backup timestamps are independent from each other so they could be done in parallel. We'll also have extra queries here if one timestamp has more result pages than the other since the one that finishes first will continue to search using the last existing sort ID.

Reply:
> the queries for primary and backup timestamps are independent from each other so they could be done in parallel.

Maybe I'm misunderstanding what you mean by "in parallel"?

Reply:
We can use `Promise.all` to start multiple queries at once so we aren't stuck waiting on the results from one query before starting the next one.

Reply:
Since we are not relying on counting the shard failures from the first searchAfter in order to run the second `@timestamp` search we can run them independently. I gotcha. Also, doesn't need to be a promise.all I could just rewrite them as

```
const singleSearchAfter = singleSearchAfter();
const singleSearchAfterB = singleSearchAfter();

const searchAfter = await singleSearchAfter;
const searchAfterB = await singleSearchAfterB;
```

right? That would enqueue them similarly to promise.all.

I think I'd like to keep the logic separate like that since we still would only want to enqueue the secondary searchAfter if there is a timestamp override field [`f0759b7` (#86368)](https://github.com/elastic/kibana/pull/86368/commits/f0759b75f3e357e8e93f515d345d2e2385e42add#diff-70a938251572c8b374514dbb063ecc7599afbb8b74aec071023c54239087662dR114)

Reply:
But yeah I can rework this to execute like the above example so that they are not running in series anymore.

Reply:
Good point, @marshallmain . In some cases, we probably can start breaking apart some larger queries to take advantage of this too... forcing multiple aggs into a single query, for example, can help to take advantage of caching, but you lose the ability to perform those aggs in parallel. Just a side note as we start to think more about our impact on scale.

Reply:
@dhurley14 That looks like that should be equivalent, yeah.